compiler/rustc_transmute/src/layout/mod.rs

```diff
@@ -62,6 +101,21 @@ impl Ref for ! {
     }
 }
 
+#[cfg(test)]
+impl<const N: usize> Ref for [u8; N] {
```

Comment:
Can you make this `[(); N]`, since the bytes do not currently encode anything?

Reply:
Or, alternatively, do what we do for `Def`: define a named type, `Ref`, that stands in for references in tests.

Reply:
Went with the `[(); N]` approach.